.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,65 @@
+name: CI/CD Workflow
+
+on:
+  push:
+    branches:
+      - 'main'
+      - 'feature/*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Install dependencies using pnpm
+        run: |
+          corepack enable
+          pnpm install
+
+      - name: Run tests
+        run: pnpm test
+
+  deploy:
+    if: github.ref == 'refs/heads/main'
+    needs: build
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '20'
+
+      - name: Install dependencies using pnpm
+        run: |
+          corepack enable
+          pnpm install
+
+      - name: Build project
+        run: pnpm run build
+
+      - name: Log in to Docker Hub
+        run: |
+          echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u "${{ vars.DOCKER_USERNAME }}" --password-stdin
+
+      - name: Build Docker image
+        run: |
+          TAG="latest"  # You can adjust the tag (e.g., based on version, commit, etc.)
+          docker build -t ${{ vars.DOCKER_USERNAME }}/my-app:$TAG .
+          docker tag ${{ vars.DOCKER_USERNAME }}/my-app:$TAG ${{ vars.DOCKER_USERNAME }}/my-app:latest
```

Comment:
I would recomment adding two tags: One for latest version (makes automated deployment easy) and one with the commit-hash, s.t the previous versions are still accessible. Otherwise it's always just overwritten.
Then of course some automated cleanup of the docker-repo is required as well

Reply:
Well latest is always there, it's the 
```yaml
TAG="latest"  # You can adjust the tag (e.g., based on version, commit, etc.)
```

That we have to adjust hmmm. In general would be awesome to have a release tags dunno how feasible that is to setup @DhanushNehru ?